esphome/components/modbus/modbus.h

```diff
@@ -28,28 +31,39 @@ class Modbus : public uart::UARTDevice, public Component {
   void register_device(ModbusDevice *device) { this->devices_.push_back(device); }
 
   float get_setup_priority() const override;
+  bool tx_buffer_empty();
+  bool tx_blocked();
 
   void send(uint8_t address, uint8_t function_code, uint16_t start_address, uint16_t number_of_entities,
             uint8_t payload_len = 0, const uint8_t *payload = nullptr);
   void send_raw(const std::vector<uint8_t> &payload);
   void set_role(ModbusRole role) { this->role = role; }
   void set_flow_control_pin(GPIOPin *flow_control_pin) { this->flow_control_pin_ = flow_control_pin; }
-  uint8_t waiting_for_response{0};
   void set_send_wait_time(uint16_t time_in_ms) { send_wait_time_ = time_in_ms; }
+  void set_turnaround_time(uint16_t time_in_ms) { turnaround_delay_ms_ = time_in_ms; }
   void set_disable_crc(bool disable_crc) { disable_crc_ = disable_crc; }
 
   ModbusRole role;
 
  protected:
   GPIOPin *flow_control_pin_{nullptr};
+  uint8_t waiting_for_response_{0};
 
   bool parse_modbus_byte_(uint8_t byte);
+  void receive_and_parse_modbus_bytes_();
   uint16_t send_wait_time_{250};
+  uint16_t frame_delay_ms_{5};
+  uint16_t long_rx_buffer_delay_ms_{0};
+  uint16_t turnaround_delay_ms_{100};
   bool disable_crc_;
   std::vector<uint8_t> rx_buffer_;
+  void clear_rx_buffer_(const std::string &reason);
   uint32_t last_modbus_byte_{0};
   uint32_t last_send_{0};
+  uint32_t last_send_tx_offset_{0};
   std::vector<ModbusDevice *> devices_;
+  std::deque<std::vector<uint8_t>> tx_buffer_;
+  void send_next_frame_();
```

Comment:
Please sort types by widest to narrowest and place pointers and other types last; this approach minimizes padding, reducing wasted bytes in RAM. I think this will work & should show a small reduction in memory use:
```suggestion
  bool parse_modbus_byte_(uint8_t byte);
  void receive_and_parse_modbus_bytes_();
  void clear_rx_buffer_(const std::string &reason);
  void send_next_frame_();

  uint32_t last_modbus_byte_{0};
  uint32_t last_send_{0};
  uint32_t last_send_tx_offset_{0};
  uint16_t frame_delay_ms_{5};
  uint16_t long_rx_buffer_delay_ms_{0};
  uint16_t send_wait_time_{250};
  uint16_t turnaround_delay_ms_{100};
  uint8_t waiting_for_response_{0};
  bool disable_crc_;

  GPIOPin *flow_control_pin_{nullptr};

  std::vector<uint8_t> rx_buffer_;
  std::vector<ModbusDevice *> devices_;
  std::deque<std::vector<uint8_t>> tx_buffer_;
```